datafusion/core/src/physical_optimizer/enforcement.rs

```diff
@@ -835,13 +836,40 @@ fn new_join_conditions(
     new_join_on
 }
 
+/// Within this function, it checks whether we need to add additional plan operators
+/// of data exchanging and data ordering to satisfy the required distribution and ordering.
+/// And we should avoid to manually add plan operators of data exchanging and data ordering in other places
 fn ensure_distribution_and_ordering(
     plan: Arc<dyn crate::physical_plan::ExecutionPlan>,
     target_partitions: usize,
 ) -> Result<Arc<dyn crate::physical_plan::ExecutionPlan>> {
     if plan.children().is_empty() {
         return Ok(plan);
     }
+    // If we have a `LIMIT` can run sort/limits in parallel (similar to TopK)
+    let plan = plan
```

Comment:
I think this comment need to be modified to reflect the new purpose.

Reply:
Thanks for the suggestion. Refined the comments.

Reply:
👍